porter/porter.go

```diff
@@ -0,0 +1,91 @@
+package porter
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+	"runtime"
+
+	"github.com/carolynvs/magex/mgx"
+	"github.com/carolynvs/magex/pkg/downloads"
+	"github.com/carolynvs/magex/shx"
+	"github.com/carolynvs/magex/xplat"
+)
+
+var (
+	// DefaultPorterVersion is the default version of Porter that is installed when it's not present
+	DefaultPorterVersion = "v1.0.0-alpha.19"
+)
+
+// Install the default version of porter
+func EnsurePorter() {
+	EnsurePorterAt(DefaultPorterVersion)
+}
+
+// Install the specified version of porter
+func EnsurePorterAt(version string) {
+	home := GetPorterHome()
+	runtimesDir := filepath.Join(home, "runtimes")
+	os.MkdirAll(runtimesDir, 0770)
+
+	clientPath := filepath.Join(home, "porter"+xplat.FileExt())
+	if _, err := os.Stat(clientPath); os.IsNotExist(err) {
```

Comment:
This ensures that porter is installed but not that the expected version is installed. Should this include version checking as well?

Reply:
Ooh yeah thanks. Previously when checking the system version installed, we only cared if it was present but didn't pitch a fit if the version was off. Now that we are controlling where it's installed, we can be more picky and require an exact version match. I'll update it.